app/components/emailPasswordAuthSignUp.tsx

```diff
@@ -1,19 +1,32 @@
 import React, { useCallback } from "react";
 import { useRouter } from "next/router";
-import { auth } from "../firebase/clientApp";
-import { createUserWithEmailAndPassword } from "firebase/auth";
+import { auth, db } from "../firebase/clientApp";
+import { updateProfile, createUserWithEmailAndPassword } from "firebase/auth";
 
 import Link from "next/link";
+import { updateDoc, doc } from "firebase/firestore";
 
 const EmailPasswordAuthSignUp = () => {
   const Router = useRouter();
 
   const signupHandler = useCallback(
     async (event: any) => {
       event.preventDefault();
-      const { email, password } = event.target.elements;
+      const { email, password, displayName } = event.target.elements;
       try {
-        await createUserWithEmailAndPassword(auth, email.value, password.value);
+        await createUserWithEmailAndPassword(auth, email.value, password.value)
+          //Update Display name for firebase auth account and firestore user document
+          .then(async (user) => {
+            await updateProfile(user.user, {
+              displayName: displayName.value,
+            });
+            await updateDoc(doc(db, "users", user.user.uid), {
```

Comment:
Don't think you need this `updateDoc`, since the user doc needs first to be created by the cloud function

Reply:
The display name for the user doc isn't written onCreate of Auth instance, and theres no way to pass in display name as an argument, so I think this is the only way to set display name for the auth instance.